starwave/starwave.py

```diff
@@ -189,12 +187,18 @@ def get_cmd(self, nstars, gr_dict, pdict):
 
         output_mags = input_mags + self.asdf_noise[idxs]
 
-        nans = np.isnan(output_mags).any(axis = 1)
+        output_bad = np.isnan(output_mags).any(axis = 1)
 
-        output_mags = output_mags[~nans]
+        output_mags = output_mags[~output_bad]
 
+        BM_out_good = np.zeros(len(BM_in_good),dtype=bool)	
+        output_bad_false_indexes = np.nonzero(~output_bad)[0]
```

Comment:
given my comments above, this would become 
`output_good_true_indexes = np.nonzero(output_good)[0]`